### PR TITLE
fix code smell and vulnerabilities

### PR DESCRIPTION
**Pull request by songleo**
Signed-off-by: Song Song Li <ssli@redhat.com>

### CODE REVIEW

File: operators/multiclusterobservability/controllers/placementrule/role.go [hunk 1]
```diff
@@ -362,7 +362,8 @@ func deleteDeprecatedRoles(c client.Client) {
 		log.Error(err, "Failed to list deprecated roles")
 		return
 	}
-	for _, role := range roleList.Items {
```
Comment (clyang82) [inline operators/multiclusterobservability/controllers/placementrule/role.go:365]: What is wrong here?

Comment (songleo) [reply operators/multiclusterobservability/controllers/placementrule/role.go:365]: Implicit memory aliasing in for loop. for example: https://husni.dev/beware-of-implicit-memory-aliasing-in-go-foor-loop/

Comment (songleo) [reply operators/multiclusterobservability/controllers/placementrule/role.go:365]: refer to https://sonarcloud.io/project/issues?id=open-cluster-management_multicluster-observability-operator&issues=AXrq11lHBBOe0YXDjbrG&open=AXrq11lHBBOe0YXDjbrG

**Pull request merged**